cns/restserver/ipam.go

```diff
@@ -182,97 +182,48 @@ func (service *HTTPRestService) GetPodIPConfigState() map[string]cns.IPConfigura
-	var (
-		resp          cns.GetPodContextResponse
-		statusCode    types.ResponseCode
-		returnMessage string
-		err           error
-	)
-
-	statusCode = types.UnexpectedError
-
-	defer func() {
-		if err != nil {
-			resp.Response.ReturnCode = statusCode
-			resp.Response.Message = returnMessage
-		}
-
-		err = service.Listener.Encode(w, &resp)
-		logger.Response(service.Name, resp, resp.Response.ReturnCode, err)
-	}()
-
-	resp.PodContext = service.GetPodIPIDByOrchestratorContext()
-}
-
-func (service *HTTPRestService) GetPodIPIDByOrchestratorContext() map[string]string {
+func (service *HTTPRestService) handleDebugPodContext(w http.ResponseWriter, r *http.Request) {
 	service.RLock()
 	defer service.RUnlock()
-	return service.PodIPIDByPodInterfaceKey
-}
-
-func (service *HTTPRestService) GetHTTPRestDataHandler(w http.ResponseWriter, r *http.Request) {
-	var (
-		resp          GetHTTPServiceDataResponse
-		returnMessage string
-		err           error
-	)
-
-	defer func() {
-		if err != nil {
-			resp.Response.ReturnCode = types.UnexpectedError
-			resp.Response.Message = returnMessage
-		}
-
-		err = service.Listener.Encode(w, &resp)
-		logger.Response(service.Name, resp, resp.Response.ReturnCode, err)
-	}()
-
-	resp.HTTPRestServiceData = service.GetHTTPStruct()
+	resp := &cns.GetPodContextResponse{
+		PodContext: service.PodIPIDByPodInterfaceKey,
+	}
+	err := service.Listener.Encode(w, resp)
+	logger.Response(service.Name, resp, resp.Response.ReturnCode, err)
 }
 
-func (service *HTTPRestService) GetHTTPStruct() HTTPRestServiceData {
+func (service *HTTPRestService) handleDebugRestData(w http.ResponseWriter, r *http.Request) {
 	service.RLock()
 	defer service.RUnlock()
-
-	return HTTPRestServiceData{
-		PodIPIDByPodInterfaceKey: service.PodIPIDByPodInterfaceKey,
-		PodIPConfigState:         service.PodIPConfigState,
-		IPAMPoolMonitor:          service.IPAMPoolMonitor.GetStateSnapshot(),
+	resp := &GetHTTPServiceDataResponse{
+		HTTPRestServiceData: HTTPRestServiceData{
+			PodIPIDByPodInterfaceKey: service.PodIPIDByPodInterfaceKey,
+			PodIPConfigState:         service.PodIPConfigState,
+			IPAMPoolMonitor:          service.IPAMPoolMonitor.GetStateSnapshot(),
+		},
 	}
+	err := service.Listener.Encode(w, resp)
+	logger.Response(service.Name, resp, resp.Response.ReturnCode, err)
 }
 
-func (service *HTTPRestService) getIPAddressesHandler(w http.ResponseWriter, r *http.Request) {
-	var (
-		req           cns.GetIPAddressesRequest
-		resp          cns.GetIPAddressStatusResponse
-		statusCode    types.ResponseCode
-		returnMessage string
-		err           error
-	)
-
-	statusCode = types.UnexpectedError
-
-	defer func() {
-		if err != nil {
-			resp.Response.ReturnCode = statusCode
-			resp.Response.Message = returnMessage
+func (service *HTTPRestService) handleDebugIPAddresses(w http.ResponseWriter, r *http.Request) {
+	req := &cns.GetIPAddressesRequest{}
+	if err := service.Listener.Decode(w, r, req); err != nil {
```

Comment:
- `return` is missing in this error block, otherwise the rest of the code still executes after a response has been sent to client
- I know its unrelated to changes, but this `Decode` function is problematic; it shouldn't be writing any output. If there are any errors, a second error json payload will also be written by `Encode` below

Reply:
fixed